### ISSUE TITLE
[Feature] option to use perl-regexp

### ISSUE THREAD

**Issue by pbrisbin** [codeclimate/codeclimate-grep#18]
I'd like to request a (pattern-specific) option to use `--perl-regexp` instead of `--extended-regexp`.

Could do `perl: true|false` or (better IMO) `regexp: fixed|basic|extended|perl` with `extended` being the default for backwards compatibility.

For context on motivation, we have a case where we'd like to flag the following:

```js
logError("Some string")
logError(someVariable)
logError(newPrefixedVariable)
```

which are all incorrect in our code-base; it's important (for error-grouping in Bugsnag) that we use:

```js
logError(new Error("something"))
```

Without a negative lookahead, it's really hard to match the bad form but not catch the good one.

**Comment by filipesperandio** [comment]
Hey @pbrisbin! Glad to see you here!
Yeah this is a nice idea indeed. I'm not doing much Quality related work lately, but wanted to 👋 

@larkinscott is currently more involved with Quality work, I think we could add this to the backlog.

**Comment by pbrisbin** [comment]
This seems to have been addressed in #19.
